test/test.js

```diff
@@ -78,28 +78,56 @@ describe('Routes', () => {
         retrievalId
       ])
       assert.strictEqual(rows.length, 0)
+      const result = {
+        success: true,
+        walletAddress,
+        startAt: new Date(),
+        statusCode: 200,
+        firstByteAt: new Date(),
+        endAt: new Date(),
+        byteLength: 100
+      }
       const updateRequest = await fetch(
         `${spark}/retrievals/${retrievalId}`,
         {
           method: 'PATCH',
           headers: { 'Content-Type': 'application/json' },
-          body: JSON.stringify({ success: true, walletAddress })
+          body: JSON.stringify(result)
         }
       )
       assert.strictEqual(updateRequest.status, 200)
-      const { rows: [updatedRetrievalRow] } = await client.query(`
+      const { rows: [retrievalResultRow] } = await client.query(`
         SELECT *
         FROM retrieval_results
         WHERE retrieval_id = $1
       `, [
         retrievalId
       ])
-      assert.strictEqual(updatedRetrievalRow.success, true)
-      assert.strictEqual(updatedRetrievalRow.wallet_address, walletAddress)
+      assert.strictEqual(retrievalResultRow.success, result.success)
+      assert.strictEqual(retrievalResultRow.wallet_address, walletAddress)
+      assert.strictEqual(
+        retrievalResultRow.start_at.getTime(),
+        result.startAt.getTime()
+      )
+      assert.strictEqual(retrievalResultRow.status_code, result.statusCode)
+      assert.strictEqual(
+        retrievalResultRow.first_byte_at.getTime(),
+        result.firstByteAt.getTime()
+      )
```

Comment:
What are the benefits of comparing `getTime` values instead of the actual `Date` values? If we get different values and the test fails, it will be difficult to understand what's different when the error message shows milliseconds since Epoch. (For example, when the time is one hour off because of timezone problems.) Have you considered using `toISOString()` or `toJSON` instead of `getTime()`?

```suggestion
      assert.strictEqual(
        retrievalResultRow.first_byte_at.toISOString(),
        result.firstByteAt.toISOString()
      )
```

This is a nitpick; feel free to ignore it.

Reply:
> What are the benefits of comparing getTime values instead of the actual Date values?

It is because strict equality checks for two `Date` objects fail even if they point at the same time, since they're different identities:

```js
> const d = new Date()
undefined
> const d2 = new Date(d.getTime())
undefined
> d === d2
false
> d
2023-05-31T13:44:34.313Z
> d2
2023-05-31T13:44:34.313Z
> 

```

We can use iso string comparison too, yeah that should be more readable